lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -50,6 +50,7 @@ export abstract class BaseManagedIdentitySource {
     protected logger: Logger;
     private nodeStorage: NodeStorage;
     private cryptoProvider: CryptoProvider;
+    private networkClient: INetworkModule;
```

Comment:
In the end, there is one and only network client. The network client may or may not perform retries. That's the beauty of the decorator pattern.

Why have 2 private variables here - networkClient and networkClientWithRetry?